packages/build/tests/steps/tests.js

```diff
@@ -41,3 +43,19 @@ test('Invalid package.json does not make build fail', async (t) => {
   const output = await new Fixture('./fixtures/invalid_package_json').runWithBuild()
   t.snapshot(normalizeOutput(output))
 })
+
+test('Build removes blobs directory before starting if there is a build command', async (t) => {
+  const fixture = await new Fixture('./fixtures/with_preexisting_blobs').withCopyRoot({ git: false })
+
+  const { success } = await fixture
+    .withFlags({
+      cwd: fixture.repositoryRoot,
+    })
+    .runBuildProgrammatic()
+
+  t.true(success)
+
+  const blobsDir = path.join(fixture.repositoryRoot, 'out', '.netlify', 'blobs', 'deploy')
+
+  await t.throwsAsync(fsPromises.access(blobsDir))
```

Comment:
Should we do this same check before the test execution (validating the dir exists)? Just to make sure we fully test the behaviour we want, having been bitten by stuff like this in the past 😅

Reply:
💯